compiler/rustc_codegen_cranelift/src/allocator.rs

```diff
@@ -134,4 +142,16 @@ fn codegen_inner(
         .define_function(func_id, &mut ctx, &mut NullTrapSink {}, &mut NullStackMapSink {})
         .unwrap();
     unwind_context.add_function(func_id, &ctx, module.isa());
+
+    let data_id = module
+        .declare_data("__rust_alloc_error_handler_should_panic", Linkage::Export, false, false)
```

Comment:
I would make the global a `sym` so that we don't have 3 places where the name can be different accidentally.

Reply:
I don't think this is an issue. We've been doing the same for other injected symbols. Plus this name is never used as a `Symbol` in the compiler so I don't it should belong there.